### PR TITLE
fix parameter not found error while checking bootstrap status

### PR DESCRIPTION
**Pull request by sobolk**
## Problem
Parameter not found handling logic was missed in initial implementation causing error instead of redirecting to boostraping page. (see https://github.com/aws-amplify/amplify-backend/pull/1576)

**Issue number, if available:**
https://github.com/aws-amplify/amplify-backend/issues/1616

## Changes

Adjusted exception handling to catch https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-ssm/Class/ParameterNotFound/ .

**Corresponding docs PR, if applicable:**

## Validation

1. Adjusted unit tests
2. Manual verification in non-boostraped region.

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/sandbox/src/file_watching_sandbox.ts [hunk 1]
```diff
@@ -325,9 +325,9 @@ export class FileWatchingSandbox extends EventEmitter implements Sandbox {
       if (
         e &&
         typeof e === 'object' &&
-        'message' in e &&
-        typeof e.message === 'string' &&
-        e.message.includes('does not exist')
+        'name' in e &&
+        typeof e.name === 'string' &&
+        e.name === 'ParameterNotFound'
```
Comment (sobolk) [inline packages/sandbox/src/file_watching_sandbox.ts:330]: see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-ssm/Class/ParameterNotFound/

Comment (Amplifiyer) [reply packages/sandbox/src/file_watching_sandbox.ts:330]: should we instead do `if (e instance of ParameterNotFound)`?

Comment (awsluja) [reply packages/sandbox/src/file_watching_sandbox.ts:330]: looks like we should in aws-sdk v3.53.0+ https://aws.amazon.com/blogs/developer/service-error-handling-modular-aws-sdk-js/

Comment (sobolk) [reply packages/sandbox/src/file_watching_sandbox.ts]: @Amplifiyer @awsluja done.

**Review by awsluja — APPROVED** [review]

File: packages/sandbox/src/file_watching_sandbox.ts [hunk 2]
```diff
@@ -322,13 +326,7 @@ export class FileWatchingSandbox extends EventEmitter implements Sandbox {
...
-        e &&
-        typeof e === 'object' &&
-        'message' in e &&
-        typeof e.message === 'string' &&
-        e.message.includes('does not exist')
-      ) {
+      if (e instanceof ParameterNotFound) {
         return false;
       }
       // If we are unable to get the stack info due to other reasons(AccessDenied), we fail fast.
```
Comment (Amplifiyer) [inline packages/sandbox/src/file_watching_sandbox.ts:332]: we should clean up this comment as well.

**Review by awsluja — APPROVED** [review]

File: packages/sandbox/src/file_watching_sandbox.ts (hunk 2 above)
Comment (sobolk) [reply packages/sandbox/src/file_watching_sandbox.ts]: done

**Review by Amplifiyer — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**